### PR TITLE
Cache collector

### PR DESCRIPTION
**Pull request by breed808**
Implements collector for Windows Cache. This is using the commits provided by @wbagdon provided in #455.
Due to the age of the PR, I've rebased the commits from current master and pushed to a new branch.

### CODE REVIEW

File: collector/cache.go [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	registerCollector("cache", newCacheCollector, "Cache")
+}
+
+// A CacheCollector is a Prometheus collector for WMI Win32_PerfFormattedData_PerfOS_Cache  metrics
```
Comment (carlpett) [inline collector/cache.go:14]: Old :)

File: collector/cache.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+func (c *CacheCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ctx, ch); err != nil {
+		log.Error("failed collecting cache metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+// Win32_PerfFormattedData_PerfOS_Cache  docs:
+// - https://docs.microsoft.com/en-us/previous-versions/aa394267(v=vs.85)
```
Comment (carlpett) [inline collector/cache.go:238]: Do we have a better link?

**Review by carlpett — APPROVED** [review]
Nice work! Looks like it addresses all the comments I had on the draft PR as well.
There are two minor nits on comments, but LGTM! :tada:

File: collector/cache.go (hunk 1 above)
Comment (breed808) [reply collector/cache.go]: Good spot, I've amended the commit to fix this.

File: collector/cache.go (hunk 2 above)
Comment (breed808) [reply collector/cache.go]: Unfortunately no. This is the only official documentation for the class I've been able to find, so @wbagdon had it right when using this link for documentation.

**Pull request merged**